pkgs/applications/window-managers/notion/default.nix

```diff
@@ -1,45 +1,47 @@
-{
-  enableXft ? true, libXft ? null, patches ? [], stdenv, lua, gettext, pkgconfig, xlibsWrapper, libXinerama, libXrandr, libX11,
-  xterm, xmessage, makeWrapper, fetchFromGitHub, mandoc, which
+{ stdenv, fetchFromGitHub, pkgconfig
+, lua, gettext, which, groff
+, xmessage, xterm
+, readline, fontconfig
+, xlibsWrapper
+, libX11, libXext, libSM
+, libXinerama, libXrandr
+, libXft
+, makeWrapper
 }:
 
-assert enableXft -> libXft != null;
-
-let
+stdenv.mkDerivation rec{
   pname = "notion";
-  version = "3-2017050501";
-  inherit patches;
-in
-stdenv.mkDerivation {
-  name = "${pname}-${version}";
-  meta = with stdenv.lib; {
-    description = "Tiling tabbed window manager, follow-on to the ion window manager";
-    homepage = http://notion.sourceforge.net;
-    platforms = platforms.linux;
-    license   = licenses.notion_lgpl;
-    maintainers = with maintainers; [jfb];
-  };
+  version = "2020-01-07";
+
   src = fetchFromGitHub {
     owner = "raboof";
     repo = pname;
-    rev = version;
-    sha256 = "1wq5ylpsw5lkbm3c2bzmx2ajlngwib30adxlqbvq4bgkaf9zjh65";
+    rev = "8d719dff20f2186a6e0c4e5cf5486da785857d5b";
+    sha256 = "1rlbhknjz4fx55lb4g7z746f32p4mxwzns244slggpr5ba51ivfg";
   };
 
-  patches = patches;
-  postPatch = ''
-    substituteInPlace system-autodetect.mk --replace '#PRELOAD_MODULES=1' 'PRELOAD_MODULES=1'
-    substituteInPlace man/Makefile --replace "nroff -man -Tlatin1" "${mandoc}/bin/mandoc -T man"
-  '';
+  nativeBuildInputs = [ pkgconfig makeWrapper groff ];
+  buildInputs = [ lua gettext which readline fontconfig
+                  libX11 libXext libSM libXft xlibsWrapper
+                  libXinerama libXrandr ];
 
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [makeWrapper xlibsWrapper lua gettext mandoc which libXinerama libXrandr libX11 ] ++ stdenv.lib.optional enableXft libXft;
+  buildFlags = [ "CC=cc" "LUA_DIR=${lua}" "X11_PREFIX=/no-such-path"
+                 # A bug in the upstream prevents it to compile
+                 # when PRELOAD_MODULES is 1.
+                 "PRELOAD_MODULES=0" ];
```

Comment:
I think this can be removed: `PRELOAD_MODULES=0` is already the default.

Reply:
Weirdly, I needed to explicitly set it as "0" or else the mysterious fontconfig DSO bug appeared.